cnf-certification-test/accesscontrol/suite.go

```diff
@@ -209,8 +209,49 @@ var _ = ginkgo.Describe(common.AccessControlTestKey, func() {
 		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
 		test1337UIDs(&env)
 	})
+
+	testID, tags = identifiers.GetGinkgoTestIDAndLabels(identifiers.TestProjectedVolumeServiceAccountIdentifier)
+	ginkgo.It(testID, ginkgo.Label(tags...), func() {
+		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
+		testProjectedVolumeServiceAccount(&env)
+	})
 })
 
+func testProjectedVolumeServiceAccount(env *provider.TestEnvironment) {
+	ginkgo.By("Testing pods to ensure they are not using projected volumes for service account access")
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+	index := 0
+	for _, put := range env.Pods {
+		for i := range put.Spec.Volumes {
+			if put.Spec.Volumes[i].Projected == nil {
+				continue
+			}
+			if put.Spec.Volumes[i].Projected.Sources == nil {
+				tnf.ClaimFilePrintf("%s, volume=%s does not use projected volumes", put, put.Spec.Volumes[i].Name)
+				continue
+			}
+			for index < len(put.Spec.Volumes[i].Projected.Sources) {
+				if put.Spec.Volumes[i].Projected.Sources[index].ServiceAccountToken != nil {
+					aContainerOut := testhelper.NewContainerReportObject(put.Spec.Volumes[i].StorageOS.VolumeNamespace,
+						put.Spec.Volumes[i].Name, put.Spec.Volumes[i].String(),
+						"use projected volumes", false).
+						SetType(testhelper.ContainerCategory).
+						AddField(testhelper.Category, put.Spec.Volumes[i].String())
+					nonCompliantObjects = append(nonCompliantObjects, aContainerOut)
+				} else {
+					aContainerOut := testhelper.NewContainerReportObject(put.Spec.Volumes[i].StorageOS.VolumeNamespace,
```

Comment:
replace line with:
aPodOut := testhelper.NewPodReportObject(put.Namespace, put.Name,
						"the projected volume Service account token field is nil",
						false).
						SetType(testhelper.ProjectedVolumeType).
						AddField(testhelper.ProjectedVolumeName, put.Spec.Volumes[i].Name).
						AddField(testhelper.ProjectedVolumeSAToken, put.Spec.Volumes[i].Projected.Sources[index].ServiceAccountToken.String())

Reply:
add the following 2 field types in testhelper.go(line 53):
	ProjectedVolumeName    = "ProjectedVolumeName"
	ProjectedVolumeSAToken = "ProjectedVolumeSAToken"

Reply:
add following object type to testhelper.go (line 63)
	ProjectedVolumeType  = "ProjectedVolume"